@b-butler any idea how to remove this warning?
```
=================================================================== warnings summary ===================================================================
md/pytest/test_nlist_tuner.py::TestMoveSize::test_attach[GPU-GradientDescent]
md/pytest/test_potential.py::test_run[GPU-Yukawa]
  /home/joaander/mambaforge/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function Simulation.__del__ at 0x7fec7035d750>
  
  Traceback (most recent call last):
    File "/home/joaander/build/hoomd/hoomd/simulation.py", line 474, in __del__
      self._operations._unschedule()
    File "/home/joaander/build/hoomd/hoomd/operations.py", line 194, in _unschedule
      self._integrator._detach()
    File "/home/joaander/build/hoomd/hoomd/operation.py", line 259, in _detach
      self._detach_hook()
    File "/home/joaander/build/hoomd/hoomd/md/integrate.py", line 60, in _detach_hook
      self._methods._unsync()
    File "/home/joaander/build/hoomd/hoomd/data/syncedlist.py", line 259, in _unsync
      self._unregister_item(item)
    File "/home/joaander/build/hoomd/hoomd/data/syncedlist.py", line 207, in _unregister_item
      value._detach()
    File "/home/joaander/build/hoomd/hoomd/operation.py", line 259, in _detach
      self._detach_hook()
    File "/home/joaander/build/hoomd/hoomd/md/methods/methods.py", line 31, in _detach_hook
      self._simulation.state.update_group_dof()
  AttributeError: 'NoneType' object has no attribute 'state'
  
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
```
It occurs every time I run, though the particular test that triggers it differs. I've seen `GPU-Yukawa`, `GPU-OPP`, and `GPU-RevCross` in 3 separate runs.

--- a/hoomd/conftest.py
+++ b/hoomd/conftest.py
@@ -195,7 +195,7 @@ def lattice_snapshot_factory(device):
                 box[2] = 0
             s.configuration.box = box
 
-            s.particles.N = numpy.product(n)
+            s.particles.N = numpy.prod(n)
             s.particles.types = particle_types
 
             if any(nx == 0 for nx in n):
@@ -717,7 +717,7 @@ class Generator:
         A value of None in shape means any length.
         """
         shape = tuple(i if i is not None else self.int(20) for i in shape)
-        return (100 * self.rng.random(numpy.product(shape))
+        return (100 * self.rng.random(numpy.prod(shape))
                 - 50).reshape(shape).astype(dtype)
 
     def variant(self):

--- a/hoomd/pytest/test_box.py
+++ b/hoomd/pytest/test_box.py
@@ -94,7 +94,7 @@ def test_scale(base_box):
 
 
 def test_volume(base_box):
-    assert isclose(base_box.volume, np.product(base_box.L))
+    assert isclose(base_box.volume, np.prod(base_box.L))
     for L in np.linspace(1, 10, 10):
         box = Box.cube(L)
         assert isclose(box.volume, L**3)

--- a/hoomd/pytest/test_local_snapshot.py
+++ b/hoomd/pytest/test_local_snapshot.py
@@ -63,7 +63,7 @@ _particle_data = dict(
     # We don't care about a valid body specification here just that we can
     # retrieve and set it correctly.
     body=dict(np_type=np.uint32,
-              value=np.linspace(-1, 10, Np, dtype=np.uint32),
+              value=np.linspace(4294967295, 10, Np, dtype=np.uint32),
               new_value=np.linspace(1, 20, Np, dtype=np.uint32),
               shape=(Np,)),
     # typeid is a signed integer in C++ despite always being nonnegative

--- a/hoomd/pytest/test_simulation.py
+++ b/hoomd/pytest/test_simulation.py
@@ -231,6 +231,9 @@ def test_state_from_gsd(device, simulation_factory, lattice_snapshot_factory,
         else:
             snapshot_dict[step] = None
 
+    if device.communicator.rank == 0:
+        f.close()
+
     for step, snap in snapshot_dict.items():
         sim = simulation_factory()
         sim.create_state_from_gsd(filename, frame=step)

